flang/lib/Optimizer/Transforms/DebugTypeGenerator.cpp

```diff
@@ -190,6 +192,28 @@ mlir::LLVM::DITypeAttr DebugTypeGenerator::convertSequenceType(
       /* associated */ nullptr);
 }
 
+mlir::LLVM::DITypeAttr DebugTypeGenerator::convertPointerLikeType(
+    mlir::Type elTy, mlir::LLVM::DIFileAttr fileAttr,
+    mlir::LLVM::DIScopeAttr scope, mlir::Location loc, bool genAllocated,
+    bool genAssociated) {
+  mlir::MLIRContext *context = module.getContext();
+
+  // Arrays and character need different treatment because DWARF have special
```

Comment:
You mention character as a special case here, but the condition just below is only checking for arrays. How are characters handled?

Reply:
The special handling of allocatable characters will come in the next PR. I only added arrays in this one as their handling was already present.